### PR TITLE
Feature/esafe 298 add rsa signing to cryppo

### PR DESCRIPTION
**Pull request by GenteiJanken**
This is intended to mirror changes to the Ruby Cryppo gem, providing a facility to sign and verify messages with an RSA keypair, as well as serialize a signature and corresponding data for transport to the keystore microservice.

### CODE REVIEW

File: src/signing/rsa-signature.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export function signWithPrivateKey(privateKeyPem: string, data: string): ISignature {
+    const mdDigest = md.sha256.create();
+    const key = pki.privateKeyFromPem(privateKeyPem) as pki.rsa.PrivateKey;
+    mdDigest.update(data, 'utf8');
+    const signature = key.sign(mdDigest);
+    return {
+        signature: signature,
+        data: data
```
Comment (zbarbuto) [inline src/signing/rsa-signature.ts:17]: Use shorthands for these

File: src/signing/rsa-signature.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export function serializeRsaSignature(signatureObj: ISignature): string{
+    return `Sign.Rsa4096.${encodeSafe64(signatureObj.signature)}.${encodeSafe64(signatureObj.data)}`;
+}
+
+export function loadRsaSignature(serializedPayload: string): ISignature{
+    const decomposedPayload = serializedPayload.split('.')
+    const signed = decomposedPayload[0]
+    const signingStrategy = decomposedPayload[1]
+    const encodedSignature = decomposedPayload[2]
+    const encodedData = decomposedPayload[3]
```
Comment (zbarbuto) [inline src/signing/rsa-signature.ts:30]: Use array destructuring for these.

**Review by zbarbuto — CHANGES_REQUESTED** [review]
Files need reformatting with prettier.

File: src/signing/rsa-signature.ts (hunk 1 above)
Comment (brewdium) [reply src/signing/rsa-signature.ts:17]: @GenteiJanken
return {
        signature,
        data

File: src/signing/rsa-signature.ts (hunk 2 above)
Comment (brewdium) [reply src/signing/rsa-signature.ts:30]: [signed, signingStrategy, encodedSignature, encodedData] = decomposedPayload

**Pull request closed without merging**